src/python/foglamp/configuration_manager.py

```diff
@@ -336,34 +365,47 @@
     """
     category_val_prepared = ''
     try:
-        # validate new category_val
+        # validate new category_val, set "value" from default
         category_val_prepared = await _validate_category_val(category_value, True)
         # check if category_name is already in storage
         category_val_storage = await _read_category_val(category_name)
         if category_val_storage is None:
-            return await _create_new_category(category_name, category_val_prepared, category_description)
+            await _create_new_category(category_name, category_val_prepared, category_description)
         else:
-            # validate category_val from storage
+            # validate category_val from storage, do not set "value" from default, reuse from storage value
             try:
                 category_val_storage = await _validate_category_val(category_val_storage, False)
+            # if validating category from storage fails, nothing to salvage from storage, use new completely
             except:
                 _logger.exception(
                     'category_value for category_name %s from storage is corrupted; using category_value without merge',
                     category_name)
+            # if validating category from storage succeeds, merge new and storage
             else:
                 category_val_prepared = await _merge_category_vals(category_val_prepared, category_val_storage, keep_original_items)
-            return await _update_category(category_name, category_val_prepared, category_description)
-
+                if json.dumps(category_val_prepared, sort_keys=True) == json.dumps(category_val_storage, sort_keys=True):
+                    # raise ValueError('category_value must be different than category_value in storage')
```

Comment:
When the value in storage is the same as the new value, I do not make an update to the DB. In these cases, should I raise an exception, or return with no message?

Reply:
If no change is made then I think it is reasonable to silently return